CMakeLists.txt

```diff
@@ -17,6 +17,10 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release" CACHE STRING "CMake build type" FORCE)
 endif()
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    add_compile_options(-fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv)
```

Comment:
let's move it to https://github.com/openvinotoolkit/openvino/blob/master/cmake/developer_package/compile_flags/sdl.cmake with detailed comments why such options are needed

Reply:
Ok, I will create a new PR.

Reply:
hi @ilya-lavrenov , New PR: https://github.com/openvinotoolkit/openvino/pull/12143 has been created, please help review. Thanks.